libcaf_io/caf/io/middleman.test.cpp

```diff
@@ -0,0 +1,26 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/main/LICENSE.
+
+#include "caf/io/middleman.hpp"
+
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/test.hpp"
+
+using namespace caf;
+using namespace caf::io;
+
+namespace {
+
+// Regression test for GH-#1900.
+// Note: we can't reproduce this in compile time,
+// static_assert(std::is_same_v<..snippet.., expected<actor>>) won't work.
+TEST("spawn client works") {
+  caf::actor_system_config cfg;
+  cfg.load<io::middleman>();
+  caf::actor_system sys{cfg};
+  sys.middleman().spawn_client(
+    [](caf::io::broker*, caf::io::connection_handle) {}, "foo", 42);
+}
```

Comment:
"spawn client works" is not a very descriptive test name. Just calling it regression test with the issue number is perfectly fine. The `connection_handle` will be interpreted as a socket internally, so just passing 42 means this code eventuall calls `closesocket(42)`, which I think we should avoid.

```
TEST("GH-1900 regression") {
  // Note: we don't need to actually run `spawn_client`, we only check for a compiler error. We shouldn't actually run this code, since it will assume 42 is a socket descriptor.
  auto fn = [] {
    caf::actor_system_config cfg;
    cfg.load<io::middleman>();
    caf::actor_system sys{cfg};
    sys.middleman().spawn_client(
      [](caf::io::broker*, caf::io::connection_handle) {}, "foo", 42);
  };
  check_eq(std::is_same_v<decltype(fn), void>);
}
```

Reply:
Good feedback, full ack.  
What's the last line about? If it's just to satisfy the testing framework, did you mean to write something like: 
```
  check(!std::is_same_v<decltype(fn), void>);
```

Reply:
Ah yeah, should be `check(std::is_same_v<decltype(fn), void>);`

This line is to have at least one check in the suite and to avoid a unused warning for `fn`.

Reply:
I've also added the `!` symbol, because this test will fail like this.

Reply:
Ah, I meant `decltype(fn())`, but doesn't really matter.